lib/widgets/button.dart

```diff
@@ -0,0 +1,169 @@
+///   * Edit-message compose box: https://www.figma.com/design/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=3988-38201&m=dev
+class ZulipWebUiKitButton extends StatelessWidget {
+  const ZulipWebUiKitButton({
+    super.key,
+    this.attention = ZulipWebUiKitButtonAttention.medium,
+    this.intent = ZulipWebUiKitButtonIntent.info,
+    required this.label,
+    required this.onPressed,
+  });
+
+  final ZulipWebUiKitButtonAttention attention;
+  final ZulipWebUiKitButtonIntent intent;
+  final String label;
+  final VoidCallback? onPressed;
+
+  WidgetStateColor _backgroundColor(DesignVariables designVariables) {
+    switch ((attention, intent)) {
+      case (ZulipWebUiKitButtonAttention.medium, ZulipWebUiKitButtonIntent.info):
+        return WidgetStateColor.fromMap({
+          WidgetState.pressed: designVariables.btnBgAttMediumIntInfoActive,
+          ~WidgetState.pressed: designVariables.btnBgAttMediumIntInfoNormal,
+        });
+      case (ZulipWebUiKitButtonAttention.high, ZulipWebUiKitButtonIntent.info):
+        return WidgetStateColor.fromMap({
+          WidgetState.pressed: designVariables.btnBgAttHighIntInfoActive,
+          ~WidgetState.pressed: designVariables.btnBgAttHighIntInfoNormal,
+        });
+    }
+  }
+
+  Color _labelColor(DesignVariables designVariables) {
+    switch ((attention, intent)) {
+      case (ZulipWebUiKitButtonAttention.medium, ZulipWebUiKitButtonIntent.info):
+        return designVariables.btnLabelAttMediumIntInfo;
+      case (ZulipWebUiKitButtonAttention.high, ZulipWebUiKitButtonIntent.info):
+        return designVariables.btnLabelAttHigh;
+    }
+  }
+
+  TextStyle _labelStyle(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    return TextStyle(
+      color: _labelColor(designVariables),
+      fontSize: 17,
+      height: 1.20,
+      letterSpacing: proportionalLetterSpacing(context,
+        0.006, baseFontSize: 17),
+    ).merge(weightVariableTextStyle(context, wght: 600));
+  }
+
+  BorderSide _borderSide(DesignVariables designVariables) {
+    switch (attention) {
+      case ZulipWebUiKitButtonAttention.medium:
+        // TODO inner shadow effect like `box-shadow: inset`, following Figma;
+        //   needs Flutter support for something like that:
+        //     https://github.com/flutter/flutter/issues/18636
+        //     https://github.com/flutter/flutter/issues/52999
+        //   For now, we just use a solid-stroke border with half the opacity
+        //   and half the width.
+        return BorderSide(
+          color: designVariables.btnShadowAttMed.withFadedAlpha(0.5),
+          width: 0.5);
+      case ZulipWebUiKitButtonAttention.high:
+        return BorderSide.none;
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return AnimatedScaleOnTap(
+      scaleEnd: 0.96,
+      duration: Duration(milliseconds: 50),
```

Comment:
Looks like duration in Figma is 100ms:

![image](https://github.com/user-attachments/assets/cd923371-e2f3-491c-be91-87a2c15ef736)

Reply:
Thanks for spotting that! I didn't see it, so instead found an animation in the web app with Chrome dev tools; probably ought to have made a comment about that :)